File: TaskPriorityMatrix/src/TaskPriorityMatrix.Core/Model/TaskAggregate/PriorityTask.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace TaskPriorityMatrix.Core;

public class PriorityTask
{
    public Guid PriorityTaskId { get; set; }

    /// <summary>
    /// Gets or sets the tenant ID for multi-tenant isolation.
    /// </summary>
    public Guid TenantId { get; set; }

    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public Urgency Urgency { get; set; }
    public Importance Importance { get; set; }
    public TaskStatus Status { get; set; } = TaskStatus.NotStarted;
    public DateTime? DueDate { get; set; }
    public Guid? CategoryId { get; set; }
    public Category? Category { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? CompletedAt { get; set; }
}

File: TaskPriorityMatrix/src/TaskPriorityMatrix.WebApp/projects/task-priority-matrix/src/app/models/priority-task.ts
import { Urgency } from './urgency';
import { Importance } from './importance';
import { TaskStatus } from './task-status';

export interface PriorityTask {
  priorityTaskId: string;
  title: string;
  description: string;
  urgency: Urgency;
  importance: Importance;
  status: TaskStatus;
  dueDate: Date | null;
  categoryId: string | null;
  createdAt: Date;
  completedAt: Date | null;
}

export interface CreatePriorityTaskRequest {
  title: string;
  description: string;
  urgency: Urgency;
  importance: Importance;
  dueDate: Date | null;
  categoryId: string | null;
}

export interface UpdatePriorityTaskRequest {
  priorityTaskId: string;
  title: string;
  description: string;
  urgency: Urgency;
  importance: Importance;
  status: TaskStatus;
  dueDate: Date | null;
  categoryId: string | null;
}
